lib/utils/index.js

```diff
@@ -605,6 +605,22 @@ class Utils {
       throw err;
     }
   }
+
+  static isSafari(desiredCapabilities = {}) {
```

Comment:
selenium 4.5 onwards `desiredCapabilities` is replaced by `capabilities`, we should look at capabilities object also I guess.

@garg3133 @gravityvi is that a correct assumption?

Reply:
Yes. Will make changes.

Reply:
> selenium 4.5 onwards desiredCapabilities is replaced by capabilities, we should look at capabilities object also I guess.

No, desiredCapabilities is not replaced by capabilities in Selenium 4.5, but what selenium-webdriver does is while creating the session, it takes desiredCapabilities object, parses it removing all the capabilities without a vendor-prefix and creates a capabilities object (which is not same as the capabilities in Nightwatch) and then sends it to the server/webdriver. So earlier, along with this capabilities object, selenium-webdriver also used to send the desiredCapabilites object from Nightwatch to the server, which it has stopped now. But no name changing has happened.

In general, desiredCapabilites contains the capabilities which are requested/desired by the user and capabilites contains the final capabilities that are used by the server and sent back to the client.

I'm not sure if what capabilities object contains in Nightwatch before the session is created, but my guess is it either contains a copy of desiredCapabilites (in which change the code change is fine) or an instance of Capabilities class of selenium-webdriver (in which case this code change can be problematic).

(Sorry for the delayed response, I meant to reply sooner but got occupied somewhere else.)

Reply:
But then same thing is being done here also, so I guess this should be fine: https://github.com/nightwatchjs/nightwatch/blob/80e30c6c0149dd3422206ddd42e755269bb6d77f/lib/transport/factory.js#L55-L60

The only thing is we are not considering if the final `desiredCapabilities` is an instance of `Capabilities` in our case. But this thing can be looked into while working on issue #3603, which is similar to this.

Reply:
There is confusing regarding how we should specify configuration in nightwatch config after selenium 4.5. The naming scheme does not help. I have created an issue to track this: https://github.com/nightwatchjs/nightwatch/issues/3626